nservicebus/sagas/index.md

```diff
@@ -22,6 +22,24 @@ Design processes with more than one remote call to use sagas.
 While it may seem excessive at first, the business implications of the system getting out of sync with the other systems it interacts with can be substantial. It's not just about exceptions that end up in the logs.
 
 
+## No saga instance available
+
+If a message that doesn't start a saga is received but no related saga instance is found, then by default that message is discarded, i.e. it is possible that a saga instance had been already completed when further messages arrive and such messages should be discarded.
+
+NOTE: Always assume that messages can be delivered out of order, i.e. due to invoking the error recovery mechanism, network latency, and/or concurrent message processing.
+
+A similar message discarding behavior can also be invoked in case of saga instance creation. When a message is received that expects a saga instance to be there but that message that would create the saga instance has not yet been received or completed processing.
+
+There are two options to avoid discarding incoming messages in this scenario:
+
+- Let the saga instance be created for the message processing that previously assumed the saga instance would already exist using `IAmStartedBy<T>`
+- Override the saga not found behavior and throw an exception using `IHandleSagaNotFound`
+
+By letting the saga instance also be created for a message that previously relied on another message that would have create the saga instance we now have to make sure that the behavior of the saga will remain correct. Lets assume that we relied on message A to be processed before B but we now also have the saga instance created when B is received then maybe state from message A needs to be included that is added to the saga instance when A is received. This state is not yet stored in the saga if message B is processed before message A thus the message cannot be send yet. We now need to check that when message A is received if message B is already processed and then have the message send. This can make the saga much more complex!
+
+A simpler solution is to throw an error when the saga instance does not exist so that it eventually gets moved to the error queue after all immediate and delayed retries attempts are done. Override saga not found behavior by [implementing `IHandleSagaNotFound` and throw an exception](saga-not-found.md). 
```

Comment:
I'm not clear - is that an alternative or complementary step? ie. we use it together with having two IAmStartedBy... messages, or instead of it?

Reply:
Instead of. So either you deal with our of order delivery by

- implementing `IAmStartedBy` for all messages that could be received out of order, manage the ordering in the saga and have a state machine that decides all messages have arrived and continue.
- rely on error recovery by implementing IHandleSagaNotFound and check the messages types that must not be discarded as a saga instance is expected to exist.